### PR TITLE
Set proper config options for HPSDR connector

### PR DESCRIPTION
**Pull request by jancona**
Add `remote` as an optional key with proper syntax. Remove `rtltcp_compat` and `iqswap` which don't apply. Make `rf_gain` a number so that it's passed as an integer value.

### CODE REVIEW

**Review by jketterl — CHANGES_REQUESTED** [review]
Looking good all round. I'd recommend removing the filter to simplify the code.

File: owrx/source/hpsdr.py [hunk 1]
```diff
@@ -34,7 +36,25 @@ def getCommandMapper(self):
...
+        )
 
 class HpsdrDeviceDescription(ConnectorDeviceDescription):
     def getName(self):
         return "HPSDR devices (Hermes / Hermes Lite 2 / Red Pitaya)"
+
+    def getInputs(self) -> List[Input]:
+        return list(filter(lambda x : x.id != "rf_gain", super().getInputs())) + [
+            RemoteInput(), NumberInput("rf_gain", "LNA Gain", 
+            "LNA gain between 0 (-12dB) and 60 (48dB)"),
```
Comment (jketterl) [inline owrx/source/hpsdr.py:52]: also, one more idea: i have implemented basic validators for inputs that should be able to make sure that all input by the user is correct. they are not used very widely yet, but if you'd like to make sure that values stay within 0 and 60, that should be the way to do it.

File: owrx/source/hpsdr.py [hunk 2]
```diff
@@ -34,7 +36,25 @@ def getCommandMapper(self):
...
+        super().__init__(
+            "remote", "Remote IP", infotext="Remote IP address to connect to."
+        )
 
 class HpsdrDeviceDescription(ConnectorDeviceDescription):
     def getName(self):
         return "HPSDR devices (Hermes / Hermes Lite 2 / Red Pitaya)"
+
+    def getInputs(self) -> List[Input]:
+        return list(filter(lambda x : x.id != "rf_gain", super().getInputs())) + [
```
[the remark this answers is not in the document]
Comment (jancona) [reply owrx/source/hpsdr.py]: I removed the filtering for getInputs()

File: owrx/source/hpsdr.py (hunk 1 above)
Comment (jancona) [reply owrx/source/hpsdr.py]: I added a RangeValidator in`owrx.form.input.validator`. It prevents entry of invalid values, but I don't see any message in the UI--the property just disappears if I hit "Apply and save" with an out-of-range value. Is that the way it's supposed to work?

Comment (jketterl) [reply owrx/source/hpsdr.py]: not really. seems like the validation system is not fit to be used with optional inputs. i'll have a look at that.

Comment (jancona) [reply owrx/source/hpsdr.py]: Even the way it is, the validation keeps the user from entering a bad value, which means the receiver will start. So it's still worthwhile.

**Review by jketterl — APPROVED** [review]

**Pull request merged**